config/env.go

```diff
@@ -140,3 +141,167 @@ func bindLegacyClientEnv() {
+		// Skip attributes that don't have the PelicanCfg_ prefix
+		if !strings.HasPrefix(attrName, "PelicanCfg_") {
+			continue
+		}
+
+		// Convert PelicanCfg_Client_PreferredCaches -> Client.PreferredCaches
+		viperKey := strings.Replace(strings.TrimPrefix(attrName, "PelicanCfg_"), "_", ".", -1)
+
+		// Evaluate the attribute to get its value
+		attrValue := ad.EvaluateAttr(attrName)
+		if attrValue.IsUndefined() {
+			log.Debugf("ClassAd attribute %s is undefined, skipping", attrName)
+			continue
+		}
+
+		// Set the value in viper based on the type
+		if attrValue.IsString() {
+			if strVal, err := attrValue.StringValue(); err == nil {
+				viper.Set(viperKey, strVal)
+				log.Debugf("Set %s = %s from job ClassAd attribute %s", viperKey, strVal, attrName)
+			} else {
+				log.Debugf("Failed to get string value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsInteger() {
+			if intVal, err := attrValue.IntValue(); err == nil {
+				viper.Set(viperKey, intVal)
+				log.Debugf("Set %s = %d from job ClassAd attribute %s", viperKey, intVal, attrName)
+			} else {
+				log.Debugf("Failed to get integer value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsBool() {
+			if boolVal, err := attrValue.BoolValue(); err == nil {
+				viper.Set(viperKey, boolVal)
+				log.Debugf("Set %s = %t from job ClassAd attribute %s", viperKey, boolVal, attrName)
+			} else {
+				log.Debugf("Failed to get boolean value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsReal() {
+			// For real numbers, get the actual float64 value
+			if realVal, err := attrValue.RealValue(); err == nil {
+				viper.Set(viperKey, realVal)
+				log.Debugf("Set %s = %f from job ClassAd attribute %s", viperKey, realVal, attrName)
+			} else {
+				log.Debugf("Failed to get real value for ClassAd attribute %s: %v", attrName, err)
+			}
+		} else if attrValue.IsList() {
+			// For lists, use ListValue() to get the list items
+			items, err := attrValue.ListValue()
+			if err != nil {
+				log.Debugf("Failed to get list value for ClassAd attribute %s: %v", attrName, err)
+				continue
+			}
+
+			// Convert list items to string slice
+			out := make([]string, len(items))
+			for i, v := range items {
+				if !v.IsString() {
+					log.Debugf("ClassAd attribute %s list element %d is %v, want string; skipping attribute", attrName, i, v.Type())
+					continue
+				}
+				out[i], _ = v.StringValue()
+			}
+			
+			viper.Set(viperKey, out)
+			log.Debugf("Set %s = %v from job ClassAd attribute %s", viperKey, out, attrName)
+		} else if attrValue.IsClassAd() {
+			// For nested ClassAds, marshal to a map structure that viper can handle
+			nestedAd, err := attrValue.ClassAdValue()
+			if err != nil {
+				log.Debugf("Failed to get ClassAd value for attribute %s: %v", attrName, err)
+				continue
+			}
+			
+			// Convert ClassAd to a map structure
+			adMap := make(map[string]interface{})
```

Comment:
You can literally marshal this to an interface.  All this custom logic is unnecessary.